src/classes/CRLP_Batch_Base.cls

```diff
@@ -335,13 +347,126 @@ public abstract class CRLP_Batch_Base {
         return pscDetailRecords;
     }
 
+    /******************************** Helper Methods *****************************/
+
+    /**
+     * @description Builds and returns batch job start query string.
+     * @param chunkLastId The last processed Id from the previous chunk.
+     * @param chunkSize The size of each chunk - used in the LIMIT statement in the query.
+     * @param fieldsToCheckForNonZero A list of fields that were checked for non-zero values,
+     * used in the query.
+     * @return The query string to be used by the batch start method.
+     */
+    @TestVisible
+    protected String getBatchStartQuery(Id chunkLastId, Integer chunkSize, List<String> fieldsToCheckForNonZero) {
+
+        BatchQueryBuilder builder = new BatchQueryBuilder(
+                this,
+                chunkLastId,
+                chunkSize,
+                fieldsToCheckForNonZero
+        );
+
+        return builder.buildQuery();
+    }
+
+    /**
+     * @description Inner Class used to build the start method query.
+     */
+    public class BatchQueryBuilder {
+        /**
+         * @description the CRLP_Batch_Base instance that created this class instance.  Enables this
+         * instance to access the member variables in the initializing class.
+         */
+        CRLP_Batch_Base base;
+        Integer chunkSize;
+        Id chunkLastId;
+        List<String> fieldsToCheckForNonZero;
+
+        BatchQueryBuilder(
+                CRLP_Batch_Base base,
+                Id chunkModeStartingId,
+                Integer chunkSize,
+                List<String> fieldsToCheckForNonZero
+        ) {
+            this.base = base;
+            this.chunkLastId = chunkModeStartingId;
+            this.chunkSize = chunkSize;
+            this.fieldsToCheckForNonZero = fieldsToCheckForNonZero;
+        }
+
+        /**
+         * @description Uses the local and base class variables to construct the batch class
+         * start query.
+         * @return The query string.
+         */
+        String buildQuery() {
```

Comment:
I really like the new implementation that uses a class to build out the batch job start query. My only comment is that this logic is unique to NonSkew mode. Long term, I'd like to see the entire start() method of the `CRLP_Batch_Base_Skew` class refactored the same way you did the `CRLP_Batch_Base_NonSkew.start()` method. However, I'm not going to push that for this release.

What I would say is that this class does belong here or possibly be abstracted out into its own class, however I think the methods should be refactored a bit to plan for the eventual implementation of a SkewMode Batch Query Builder. The `buildQuery` method already has a simple switch that splits between the two modes. What I would do is abstract this out further so that all the logic in the `when NonSkewMode` block is in a method or methods of its own. This way the `buildQuery` method is small and easy to follow.

It's also possible that some logic can be raised up or down a level, such as checking for `if (base.hasActiveRollupsForThisJob() == false)` to determine if the query should even return any records at all.

Reply:
We could consider moving the BatchQueryBuilder out to its own top-level class, but I kind of like storing within the base class itself.  Just kind of feels like the right place to me, all self-contained here instead of in another file/class.  I added a specific method for `buildQueryForNonSkewJob` so that later on we can add a similar one for Skew jobs.